src/prompt/base.rs

```diff
@@ -97,6 +97,10 @@ pub trait Prompt: Send {
         &self,
         history_search: PromptHistorySearch,
     ) -> Cow<str>;
+    /// Whether to repaint the prompt after the user hits enter (for transient prompts)
+    fn repaint_on_enter(&self) -> bool {
+        false
+    }
```

Comment:
Mhh I don't like this as a silently stateful API.

I would prefer this to be explicit. Three ideas that come to mind:

- making this an *explicit* setter, by preferrably having `&mut self` and either a descriptive name or an additional parameter setting the post submit state. Could still have No-Op default impl.
- passing the information that the prompt should take its post submit form directly to the `render_*` methods as a parameter. This has the downside of being a breaking change and increasing the API beginners have to learn
- adding additional `render_post_submit_*` methods that have a default implementation which would just invoke their regular form, but could be overridden by the more complex `Prompt` implementations.